app/src/androidTest/java/com/github/leuludyha/ibdb/BookSearchTest.kt

```diff
@@ -0,0 +1,188 @@
+        }
+
+        composeTestRule.waitForIdle()
+
+        composeTestRule
+            .onNodeWithTag("book_search::search_field")
+            .performTextInput("Blabla")
+
+        composeTestRule
+            .onNodeWithTag("book_search::search_field")
+            .performImeAction()
+
+        composeTestRule.waitUntil(timeoutMillis = 1000) {
+            result != null
+        }
+
+        assertEquals(listOf(Mocks.work), result)
+    }
+
+    //I am not sure that this is a good test returning that Result.Loading doesn't look ideal to me
+    @Test
+    fun queryLoadingIndicatorIsDisplayed() {
+        composeTestRule.setContent {
+            //No need for a (working) navigator in this test
+            val navController = rememberNavController()
+
+            BookSearch(
+                navController = navController,
+                outerPadding = PaddingValues(),
+                viewModel = BookSearchViewModel(
+                    SearchUseCase(LibraryRepositoryMock {
+                        return@LibraryRepositoryMock Result.Loading()
+                    })
+
+                )
+            ) {
+                //Does nothing for this test
+            }
+        }
+
+        composeTestRule.waitForIdle()
+
+        composeTestRule
+            .onNodeWithTag("book_search::query_loading_indicator")
+            .assertDoesNotExist()
+
+        composeTestRule
+            .onNodeWithTag("book_search::search_field")
+            .performTextInput("Blabla")
+
+        composeTestRule
+            .onNodeWithTag("book_search::search_field")
+            .performImeAction()
+
+        composeTestRule.waitForIdle()
+
+        composeTestRule
+            .onNodeWithTag("book_search::query_loading_indicator")
+            .assertExists()
+    }
+}
+
+/**
+ * This mock class is created to control the kind of requests the view model will make.
+ */
+private class LibraryRepositoryMock(val searchReturn: () -> Result<List<Work>> = { Result.Success(listOf(Mocks.work)) }) : LibraryRepository {
+    override suspend fun search(query: String): Result<List<Work>> {
+        return searchReturn.invoke()
+    }
+
+    override suspend fun workById(workId: String): Result<Work> {
+        return Result.Success(Mocks.work)
+    }
+
+    override suspend fun worksByAuthorId(authorId: String): Result<List<Work>> {
+        return Result.Success(listOf(Mocks.work))
+    }
+
+    override suspend fun editionsByWorkId(workId: String): Result<List<Edition>> {
+        return Result.Error("Couldn't retrieve any edition")
+    }
+
+    override suspend fun editionById(editionId: String): Result<Edition> {
+        return Result.Error("Couldn't retrieve any edition")
+    }
+
+    override suspend fun editionByISBN(isbn: Long): Result<Edition> {
+        return Result.Error("Couldn't retrieve any edition")
+    }
+
+    override suspend fun authorById(authorId: String): Result<Author> {
+        return Result.Success(Mocks.author)
+    }
+}
```

Comment:
Very good suggestion, once I have time I will add more default parameters so that it can be quickly tuned for tests modifying only the necessary methods and not all of them, as it already does for search.